File: 백준/Silver/2839. 설탕 배달/설탕 배달.py
import sys
X=int(sys.stdin.readline())
a=X%5
b=int(X/5)
while a%3!=0:
    a=a+5
    b=b-1
    if b<0:
        break;

if b<0:
    print("-1")
else:
    print(int(a/3)+b)
    

File: 백준/Silver/2839. 설탕 배달/설탕 배달.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;



public class Main {

	static int N;
	static int[] arr = new int[]{0,-1,-1,1,-1,1,2,-1,2,3,2,3,4,3,4};
	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		//while(true){
			N = Integer.parseInt(br.readLine());
			if(N<15){
				System.out.println(arr[N]);
				return;
			}
			int five = (N/5);
			N%=5;
			for(int i =0; i<3; i++){
				if(arr[N+5*i]!=-1){
					System.out.println(five+arr[N+5*i]-i);
					return;
				}
			}
			System.out.println(-1);
		//}
	}
}